### ISSUE TITLE
Missed OpSwitch

### ISSUE THREAD

**Issue by iliya-diyachkov** [KhronosGroup/LLVM-SPIRV-Backend#18]
There is missed OpSwitch in 4 tests:
```
transcoding/OpSwitchEmpty.ll
transcoding/OpSwitchChar.ll
transcoding/OpSwitch64.ll
transcoding/OpSwitch32.ll
```

**Comment by iliya-diyachkov** [comment]
Currently it's transformed to a set of compares and  conditional branches like this:
```
  switch i32 %0, label %sw.epilog [
    i32 0, label %sw.bb
    i32 1, label %sw.bb1
  ]
```
```
  %19:id = OpIEqual %51:type, %18:id, %12:id
  OpBranchConditional %19:id, %60:id, %59:id

bb.5.entry:
; predecessors: %bb.1
  successors: %bb.3, %bb.4

  %51:type = OpTypeBool
  %59:id = OpLabel
  OpName %59:id, 1920233061, 121
  %21:id = OpIEqual %51:type, %18:id, %20:id
  OpBranchConditional %21:id, %61:id, %62:id
```

**Comment by zainja** [comment]
I have a solution for this but it will not be accepted because it modifies the IRTranslator there is no other way to deal with it

**Comment by iliya-diyachkov** [comment]
Yes, let's wait till Alexander upload his changes in it.

**Comment by iliya-diyachkov** [comment]
Perhaps we need to introduce one more intrinsic which will keep the switch info thru IRTranslator.

**Comment by zuban32** [comment]
I've already looked into this, and the problem is that we can't pass labels (i.e. BBs) to the intrinsics directly. The viable solution that first comes to mind is using some basic blocks numbering, but that would work only if there're no control flow changes between pre-legalization and instruction selection. I suppose that's true for now, not sure if that'd be required in future.

**Comment by iliya-diyachkov** [comment]
Fixed in #146.
